src/Utilities/CssDocument.php

```diff
@@ -82,23 +61,45 @@ public function __construct(string $css)
      */
     public function getStyleRulesData(array $allowedMediaTypes): array
     {
-        $splitCss = $this->splitCssAndMediaQuery($allowedMediaTypes);
-
         $ruleMatches = [];
-        foreach ($splitCss as $cssPart) {
-            // process each part for selectors and definitions
-            \preg_match_all('/(?:^|[\\s^{}]*)([^{]+){([^}]*)}/mi', $cssPart['css'], $matches, PREG_SET_ORDER);
-
-            foreach ($matches as $cssRule) {
+        /** @var CssRenderable $rule */
+        foreach ($this->sabberwormCssDocument->getContents() as $rule) {
+            if ($rule instanceof CssAtRuleBlockList) {
+                $media = $this->getFilteredAtIdentifierAndRule($rule, $allowedMediaTypes);
+                if ($media !== null) {
```

Comment:
I'd prefer `is_string` here instead so we can communicate what it should be, not what it shouldn't be.

Reply:
Fair point, well made.  Changed.